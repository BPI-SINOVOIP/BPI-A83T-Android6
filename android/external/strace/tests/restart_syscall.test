#!/bin/sh

# Check how "resuming interrupted nanosleep" works.

. "${srcdir=.}/init.sh"

check_prog sleep
check_prog grep
run_strace -enanosleep sleep 1
grep nanosleep < "$LOG" > /dev/null ||
	framework_skip_ 'sleep does not use nanosleep'

./set_ptracer_any sleep 2 &
sleep 1
run_strace -q -erestart_syscall -p $!

case "$STRACE_ARCH" in
	alpha|mips|or1k|s390|s390x)
		# These architectures use the same register
		# both for syscall number and syscall return code.
		match_diff "$LOG" "$srcdir/restart_syscall_unknown.expected"
		;;
	*) match_diff
		;;
esac
