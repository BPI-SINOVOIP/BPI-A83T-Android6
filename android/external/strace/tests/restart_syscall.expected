restart_syscall(<... resuming interrupted nanosleep ...>) = 0
+++ exited with 0 +++
